### ISSUE TITLE
Bluebird / Promise.js warning of deprecated Promise.defer from CircleCi

### ISSUE THREAD

**Issue by ChaseBig** [cypress-io/cypress#5571]
## Issue Description
When running tests in CircleCi, tests failures are no longer uploading to my cypress project dashboard. The end of the cypress test run command step in CircleCi shows the message block shown below. The last recording showing on my dashboard was roughly 24hrs ago and no new test-failure recordings have been uploaded since. 
``` 
Recorded Run: https://dashboard.cypress.io/#/projects/<I-removed-recording-projectId>/runs/2464                               

(node:987) Warning: a promise was created in a handler at internal/timers.js:443:21 but was not returned from it, see http://goo.gl/rRqMUw
    at Function.Promise.cast (/root/.cache/Cypress/3.5.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/promise.js:225:13)
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Exited with code 2
```

Other test warnings are being shown as well, but these are not causing failed upload operation. 
```
(node:987) Warning: Promise.defer is deprecated and will be removed in a future version. Use new Promise instead.
``` 
and
```
tput: No value for $TERM and no -T specified` 
```

### Desired behavior:

After tests run, any failed tests should automatically upload the test recording to my cypress project dashboard.

### Steps to reproduce: (app code and test code)

I don't know what the source of the error is so I don't know how to reproduce it other than kicking off another build in CircleCi. 

This is the contents of my support/index.js code. I have added these in an attempt to debug the error I'm having based on various other reported issues. ( https://github.com/cypress-io/cypress/issues/874 )

support/index.js
```js
// Import commands.js using ES2015 syntax:
import './commands'
// require('cypress-xpath')

Cypress.on('window:before:load', win => {
  win.indexedDB.deleteDatabase('localforage')
})

Cypress.on('uncaught:exception', (err, runnable) => {
  console.log(err, runnable)
  return false
})

Cypress.on('fail', err => {
  debugger
})

// patch Cypress top.onerror
Object.defineProperty(top, 'onerror', {
  value: window.onerror
})
```

### Versions

Cypress: v3.5.0
Chrome: Most updated version available
Electron: Most updated version available

**Comment by jennifer-shehane** [comment]
Hey @ChaseBig I know you tried updating to 3.6, did this resolve your issue?

Also, we don't use `Promise.defer` in our code, could you possibly be using a version of BlueBird Promise yourself that may be throwing this warning?

I see you're using `yarn` and I know sometimes it tries to flatten our versions. This is purely speculative, but maybe there is something strange going on with the Bluebird versions between your own and Cypress?

Let us know if this sounds at all possible.

**Comment by ChaseBig** [comment]
I'm looking into the Bluebird stuff with my dev. Is there anything I can add to the index.js file to bypass bluebird errors?

**Comment by oak-wildwood** [comment]
You should be able to use [this technique](https://on.cypress.io/catalog-of-events#To-catch-a-single-uncaught-exception) to bypass this error. Just update the `.includes()` argument.

**Comment by Hooked74** [comment]
first warning `(node: 55218) Warning: Promise.defer is deprecated and will be removed in a future version. Use new Promise instead.)` does not appear due to the use of Promise.defer, but because of the use of Promise.pending, which used in **video_capture.js:70:23**.

```js
// bluebird/js/release/promise.js:758:13
Promise.defer = Promise.pending = function() {
    debug.deprecated("Promise.defer", "new Promise");
    var promise = new Promise(INTERNAL);
    return {
        promise: promise,
        resolve: deferResolve,
        reject: deferReject
    };
};
```

Stack trace:
```
    at Object.trace (internal/console/constructor.js:338:11)
    at Function.Promise.defer.Promise.pending (/Users/igor/Library/Caches/Cypress/3.6.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:758:13)
    at Object.start (/Users/igor/Library/Caches/Cypress/3.6.0/Cypress.app/Contents/Resources/app/packages/server/lib/video_capture.js:70:23)
    at /Users/igor/Library/Caches/Cypress/3.6.0/Cypress.app/Contents/Resources/app/packages/server/lib/modes/run.js:632:6
    at tryCatcher (/Users/igor/Library/Caches/Cypress/3.6.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/Users/igor/Library/Caches/Cypress/3.6.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:547:31)
    at Promise._settlePromise (/Users/igor/Library/Caches/Cypress/3.6.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:604:18)
    at Promise._settlePromise0 (/Users/igor/Library/Caches/Cypress/3.6.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:649:10)
    at Promise._settlePromises (/Users/igor/Library/Caches/Cypress/3.6.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:729:18)
    at Promise._fulfill (/Users/igor/Library/Caches/Cypress/3.6.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:673:18)
    at /Users/igor/Library/Caches/Cypress/3.6.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/nodeback.js:42:21
    at /Users/igor/Library/Caches/Cypress/3.6.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/fs-extra/lib/mkdirs/mkdirs.js:56:16
    at callback (/Users/igor/Library/Caches/Cypress/3.6.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/graceful-fs/polyfills.js:295:20)
    at FSReqCallback.oncomplete (fs.js:164:5)
```

`(node: 56453) Warning: a promise was created in a handler at internal/timers.js:443:21 but was not returned from it, see http://goo.gl/rRqMUw)`

Stack trace:
```
    at Object.trace (internal/console/constructor.js:338:11)
    at Object.checkForgottenReturns (/Users/igor/Library/Caches/Cypress/3.6.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/debuggability.js:616:15)
    at Promise._settlePromiseFromHandler (/Users/igor/Library/Caches/Cypress/3.6.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:558:15)
    at Promise._settlePromise (/Users/igor/Library/Caches/Cypress/3.6.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:604:18)
    at Promise._settlePromise0 (/Users/igor/Library/Caches/Cypress/3.6.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:649:10)
    at Promise._settlePromises (/Users/igor/Library/Caches/Cypress/3.6.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:729:18)
    at _drainQueueStep (/Users/igor/Library/Caches/Cypress/3.6.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/async.js:93:12)
    at _drainQueue (/Users/igor/Library/Caches/Cypress/3.6.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/async.js:86:9)
    at Async._drainQueues (/Users/igor/Library/Caches/Cypress/3.6.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/async.js:102:5)
    at Immediate.Async.drainQueues [as _onImmediate] (/Users/igor/Library/Caches/Cypress/3.6.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/async.js:15:14)
    at processImmediate (internal/timers.js:443:21)
```

**Comment by jennifer-shehane** [comment]
@Hooked74 Thanks! This is helpful. I'm still seeing this as only a Warning. Is this verified to cause the crashing?

**Comment by ChaseBig** [comment]
@jennifer-shehane  @oak-wildwood 
Thank you for taking time to review this issue and thank you @Hooked74 for following up with a _much_ better explanation. Jennifer is right that these are just warnings and, while they're a bit annoying to see, they're harmless as far as I can tell. I was able to do more digging into my build log and the failures I've been experiencing are related to `xvfb`. I run my cypress tests a little differently than I'd imagine most do and believe that my issues will resolve once I get multiple docker images set up. We have a monorepo architecture and instead of putting all my tests in the cypress directory at the root I installed cypress onto three of our webApp packages. My build looks for package changes in those three webApp packages and spins up local environments for any of the packages that changes are found and then I run cypress tests against those changed packages. I'm guessing that my single docker image in circleCi is getting overloaded and causing xvfb errors.

**Comment by jennifer-shehane** [comment]
@ChaseBig Great, thanks for detailing. We'll close this issue as soon as this PR can be merged in, removing the Promise.pending. https://github.com/cypress-io/cypress/pull/5611

There are some debugging tips for xvfb in our docs here: https://on.cypress.io/continuous-integration#Xvfb

**Comment by ChaseBig** [comment]
Wonderful! Thank you so much, @jennifer-shehane!

**Comment by jennifer-shehane** [comment]
Released in [3.6.1](https://on.cypress.io/changelog#3-6-1)
